apis/proto/v1alpha1/ess.proto

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2023 The Crossplane Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+option go_package = "ess-grpc-2/proto/ess";
+
+// This service defines the APIs for an External Secret Store plugin.
+service ExternalSecretStoreService {
+  rpc GetSecret(GetSecretRequest) returns (GetSecretResponse) {}
+  rpc ApplySecret(ApplySecretRequest) returns (ApplySecretResponse) {}
+  rpc DeleteKeys(DeleteKeysRequest) returns (DeleteKeysResponse) {}
+}
+
+message ConfigReference {
+  string api_version = 1;
+  string kind = 2;
+  string name = 3;
+}
```

Comment:
Is the idea that the plugin will connect to Kubernetes to read the config? Perhaps it would make sense to have plugins be unaware of Kubernetes - i.e. to have the caller read the config and pass it to the plugin via gRPC.

Reply:
One scenario we considered was, deploying the plugin and the ESS to a different cluster where XP is running.  In that case, the `vaultconfig` CRD and the config resource will be installed in the ESS cluster and it won't be possible for clients (XP and providers) to read and send it to the gRPC server.

Reply:
While it sounds appealing to make the plugin unaware of Kubernetes, it introducing its own CRD, e.g. Vault specific configuration, makes it infeasible. The scenario that @ezgidemirel mentioned is a hypothetical example of this.

Reply:
I see. Thanks for the context!